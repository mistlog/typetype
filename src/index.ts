export * from "./parser";
export * from "./generator";
export * from "./core";