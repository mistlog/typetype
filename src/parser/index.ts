export * from "./common";
export * from "./basic";
export * from "./function";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";