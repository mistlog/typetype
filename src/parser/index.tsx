export * from "./common";
export * from "./basic";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";