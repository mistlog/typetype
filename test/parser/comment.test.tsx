import { ReactPeg } from "react-peg";
import { MultiLineComment, _ } from "../../src";

describe("parser: comment", () => {
    test("MultiLineComment", () => {
        const parser = ReactPeg.render(<MultiLineComment />);
        expect(parser.parse("/* this is comment */")).toMatchSnapshot();
    })

    test("MultiLineComment: 2", () => {
        const parser = ReactPeg.render(<_ />);
        expect(parser.parse(`
            /* 
                this is comment 
            */
        `)).toMatchSnapshot();
    })
    
    test("MultiLineComment: 3", () => {
        const parser = ReactPeg.render(<_ />);
        expect(parser.parse(`
            /**
             * this is comment
             */
        `)).toMatchSnapshot();
    })
})