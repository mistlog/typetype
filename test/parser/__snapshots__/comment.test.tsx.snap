// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser: comment MultiLineComment 1`] = `
Object {
  "comment": " this is comment ",
}
`;

exports[`parser: comment MultiLineComment: 2 1`] = `
Array [
  "
",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  Object {
    "comment": " 
                this is comment 
            ",
  },
  "
",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
  " ",
]
`;
